bundler/lib/bundler/lockfile_parser.rb

```diff
@@ -110,21 +110,8 @@ def may_include_redundant_platform_specific_gems?
     def parse_source(line)
       case line
       when SPECS
-        case @type
-        when PATH
-          @current_source = TYPES[@type].from_lock(@opts)
-          @sources << @current_source
-        when GIT
-          @current_source = TYPES[@type].from_lock(@opts)
-          @sources << @current_source
-        when GEM
-          @opts["remotes"] = Array(@opts.delete("remote")).reverse
-          @current_source = TYPES[@type].from_lock(@opts)
-          @sources << @current_source
-        when PLUGIN
-          @current_source = Plugin.source_from_lock(@opts)
-          @sources << @current_source
-        end
+        @current_source = TYPES[@type]&.from_lock(@opts)
+        @sources << @current_source if @current_source
```

Comment:
Since there's no else on the former  case statement, this should be the following so the instance vars are not overwritten on an unrecognized type.

```suggestion
        return unless TYPES.key?(@type)
        @current_source = TYPES[@type].from_lock(@opts)
        @sources << @current_source
```